### PR TITLE
JAMES-2586 reworking attachmentId

### PR DESCRIPTION
**Pull request by hungphan227**
https://github.com/apache/james-project/pull/2142/commits/666ca46aed021db37968f5ebb0160b7f9c0a344a ([JAMES-2586 Create AttachmentIdFactory](https://github.com/apache/james-project/pull/2142/commits/666ca46aed021db37968f5ebb0160b7f9c0a344a)) has been reviewed in https://github.com/apache/james-project/pull/2133

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Comment by Arsnael** [comment]
Please rebase

**Comment by Arsnael** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-2142/3/#showFailuresLink

Lots of errors, can you check?

File: server/container/guice/protocols/jmap/src/main/java/org/apache/james/jmap/rfc8621/RFC8621MethodsModule.java [hunk 1]
```diff
@@ -198,7 +197,6 @@ protected void configure() {
         blobResolverMultibinder.addBinding().to(MessageBlobResolver.class);
         blobResolverMultibinder.addBinding().to(UploadResolver.class);
         blobResolverMultibinder.addBinding().to(MessagePartBlobResolver.class);
-        blobResolverMultibinder.addBinding().to(AttachmentBlobResolver.class);
```
Comment (Arsnael) [inline server/container/guice/protocols/jmap/src/main/java/org/apache/james/jmap/rfc8621/RFC8621MethodsModule.java:201]: Why is that necessary?

Comment (Arsnael) [reply server/container/guice/protocols/jmap/src/main/java/org/apache/james/jmap/rfc8621/RFC8621MethodsModule.java:201]: => https://github.com/apache/james-project/pull/2150 

got the answer :)

**Review by Arsnael — APPROVED** [review]

**Review by vttranlina — APPROVED** [review]
